rustls/src/error.rs

```diff
@@ -262,6 +262,19 @@ pub enum CertificateError {
     Other(Arc<dyn StdError + Send + Sync>),
 }
 
+impl From<CertificateError> for AlertDescription {
+    fn from(e: CertificateError) -> Self {
+        use CertificateError::*;
+        match e {
+            BadEncoding | UnhandledCriticalExtension | NotValidForName => Self::BadCertificate,
```

Comment:
Should we turn `UnhandledCriticalExtension` into `UnsupportedExtension`?

Reply:
Following the TLS1.2 [RFC](https://www.rfc-editor.org/rfc/rfc5246#section-7.2.2)

>    unsupported_extension
      sent by clients that receive an extended server hello containing
      an extension that they did not put in the corresponding client
      hello.  This message is always fatal.

And TLS1.3 [RFC](https://www.rfc-editor.org/rfc/rfc8446#section-6.2)

>    unsupported_extension:  Sent by endpoints receiving any handshake
      message containing an extension known to be prohibited for
      inclusion in the given handshake message, or including any
      extensions in a ServerHello or Certificate not first offered in
      the corresponding ClientHello or CertificateRequest.

I think it means only the extension in TLS, not certificate extension.

Reply:
Great!